bin/main.rb

```diff
@@ -1,58 +1,106 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
-puts 'Hello World!'
+class Player
+  attr_reader :name
+  attr_reader :sym
 
-# options for movement
+  def initialize(sym, name)
+    @sym = sym
+    @name = name
+  end
+end
 
-square_spaces = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+class Board
+  def initialize
+    @state = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+    show # try
+  end
+
+  def available
+    @state.each_with_index { |x, index| print "#{index + 1} " if x == ' ' }
+  end
 
-def aviable_squares
-  check = true
-  while check
-    avi = []
-    square_spaces.each_with_index do |index, space|
-      avi << index if space == ' '
+  def position(num_pos, sym)
+    check = true
+    move = num_pos.to_i
+    while check
+      if move <= 9 && move >= 1 && (move.is_a? Integer) && @state[move - 1] == ' '
+        @state[move - 1] = sym
+        check = false
+      else
+        puts 'Incorrect value, please type a correct one.'
+        available
+        puts 'are the correct numbers'
+        move = gets.chomp.to_i
+      end
     end
-    move = gets.chomp
-    if move <= 9 && move >= 1 && (move.is_a? Integer) && square_spaces[move - 1] == ' '
-      check = false
-    else
-      puts 'wrong input'
+    show
+  end
+
+  # method to print the table every iteration
+  def show
+    puts "#{@state[0]} | #{@state[1]} | #{@state[2]}"
+    puts '---------'
+    puts "#{@state[3]} | #{@state[4]} | #{@state[5]}"
+    puts '---------'
+    puts "#{@state[6]} | #{@state[7]} | #{@state[8]}"
```

Comment:
**[REQUIRED]**
All the user interaction code should not reside inside the logic classes. Eg `puts` and `gets.chomp`. The should only be included in the user interactive code in the `main.rb`, since we need to move the classes to the `lib` folder. [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/main_rb.png) from the [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/).